### DISCUSSION TITLE
Your favorite way to inspect MBTiles

### DISCUSSION THREAD

**Discussion by roblabs** [maplibre/maplibre#207] [Polls]
A Poll / survey of your favorite MBTiles validation or inspection or viewing tool.  

Key question:  _Before you go to production, how do you validate the data in your MBTile?_

I'm putting together some documentation for my team on the various tools that can be used to inspect MBTiles.  I thought I would share out my inventory of tools I've used over the years.  I came from the printing industry, so we would also call this _soft proofing_ (where hard proof would've been a printed page).  

 An MBTile is an [SQLite](https://github.com/mapbox/mbtiles-spec) database, so that opens up the options beyond map viewers.  For example, you can use `sqlite3 data.mbtiles`.

#### `ogrinfo`

Part of the [GDAL](https://gdal.org) command line tools.

```bash
# dump metadata
ogrinfo pt4_glacier_tippe.mbtiles
```

> INFO: Open of `data.mbtiles' using driver `MBTiles' successful.
Metadata:
  ZOOM_LEVEL=8
  version=2
  minzoom=0
  maxzoom=8
  center=-141.328125,61.268414,8
  bounds=-180.000000,-85.051129,180.000000,83.557563
  type=overlay
  format=pbf
  generator=tippecanoe v2.1.0

---

#### `mb-util`

A [python app](https://github.com/mapbox/mbutil) you can use to extract protobuf or inspect metadata

```bash
# dump metadata
mb-util data.mbtiles dumps
```

> Exporting MBTiles metatdata from data.mbtiles
{
  "version": "2",
  "minzoom": "0",
  "maxzoom": "15",
  "center": "-77.041626,38.912407,15",
  "bounds": "-170.773790,-47.239923,145.731763,82.782701",
  "format": "pbf",
  "generator": "tippecanoe v2.1.0",
}

---

#### MapTiler Engine

Simply open your MBTile in a GUI from our friends at MapTiler.  MapTiler Desktop is now [MapTiler Engine](https://www.maptiler.com/engine/).

<img width=31.6% src=https://user-images.githubusercontent.com/118112/215195253-ae95e9cf-b357-4c96-9800-8637a3ccd24a.jpeg>

---

#### QGIS

Simply drag MBTile into [QGIS](https://qgis.org).

<img width=31.6% src=https://user-images.githubusercontent.com/118112/215183570-4da2d48a-64e1-45e7-9130-bd7e901b7da8.jpeg>

---

#### TileServer GL

Configure [TileServer GL](https://maptiler-tileserver.readthedocs.io) to serve MBTiles with custom styles.

<img width=31.6% src=https://user-images.githubusercontent.com/118112/215191398-11600469-1b26-4d69-a9af-31c50f5b3df2.png>

---

#### consbio/mbtileserver

Alternative to tileserver, written in Go, also uses maplibre frontend: https://github.com/consbio/mbtileserver

---
#### SQLite Browser

[DB Browser for SQLite](https://sqlitebrowser.org) — an open source, freeware visual tool used to create, design and edit SQLite database files.  MBTiles 

<img width=31.6% src=https://user-images.githubusercontent.com/118112/215191572-27a6200b-f485-418f-99d7-e1ff5f5752ba.jpeg>

---

#### Extracting & Serving raw vector or rasters

You can extract the [raw vector](https://github.com/maplibre/demotiles/tree/gh-pages/tiles-omt) or raster data, and serve it any way you like.  Store in the cloud, or [preload protobuf](https://github.com/roblabs/openmaptiles-ios-demo/tree/master/OSM2VectorTiles/geography-class.osm2vectortiles) or raster tiles on a mobile device.

---

#### Tippecanoe

The Tippecanoe tools: https://github.com/felt/tippecanoe/#tippecanoe-enumerate

---

**Comment by maxammann** [comment]
Martin is missing in the list! :D

**Reply by roblabs** [reply]
Excellent!!  Can you add a link to Martin so that future viewers of this discussion can make use of Martin.

I added the tools that I used, but I know there are others!

**Comment by acalcutt** [comment]
I voted TileServer-GL because the included inspect is how I most often check out new files in my project and other generated map files. However I've used most of the free utilies on this list 

ogrinfo/ogr2ogr (converting shape files?)
mb-util (extracting/compressing mbtiles)
SQLite Browser (viewing/editing metadata)
QGIS (visualizing lake shape files and lake lines)
Tippecanoe (I use this to convert millions of gps points into a useable mbtiles file)
GDAL(not listed - so many tools + info tools)
MapTiler Engine - I've used some of the free tools, but a lot of my files are too big for the free versions...sad...

Really they all have their uses...so it was a hard vote...

**Reply by acalcutt** [reply]
Also, you could add the maplibre-gl-inspect plugin to the list
https://github.com/acalcutt/maplibre-gl-inspect

This is the same thing tileserver-gl uses, but it can be used as a stand alone plugin too.

**Reply by roblabs** [reply]
For inspection type tools, you have to hand it to Mapeture from Stamen.  MapTiler engine also has this feature, and it's my first choice usually.

* https://github.com/stamen/maperture
* https://stamen.github.io/maperture

I have also used the SQL browser for MBTiles packed with raster tiles, as it has a basic image viewer.  

Sadly, the GitHub discussion only allows 10 entries and one "choice".  I was hoping for _multinomial_ choices :)
